src/plugins/event-plugins/WebVitalsPlugin.ts

```diff
@@ -34,28 +47,63 @@ export class WebVitalsPlugin extends InternalPlugin {
     configure(config: any): void {}
 
     protected onload(): void {
+        this.context.eventBus.subscribe(Topic.EVENT, this.handleEvent); // eslint-disable-line @typescript-eslint/unbound-method
         onLCP((metric) => this.handleLCP(metric));
         onFID((metric) => this.handleFID(metric));
         onCLS((metric) => this.handleCLS(metric));
     }
 
-    handleLCP(metric: LCPMetricWithAttribution | Metric) {
+    private handleEvent = (event: ParsedRumEvent) => {
+        switch (event.type) {
+            // lcp resource is either image or text
+            case PERFORMANCE_RESOURCE_EVENT_TYPE:
+                const details = event.details as ResourceEvent;
+                if (
+                    details.fileType === ResourceType.IMAGE ||
+                    details.initiatorType === 'img'
```

Comment:
suggestion: Use the shared `getResourceFileType` function (from common-utils.ts) here to help ensure file type decisions are consistent.

Reply:
fileType is a RUM concept not included in the PerformanceResourceTiming schema, so checking fileType is a check on `getResourceFileType`.

Reply:
I removed the check to `initiatorType`